cupy/polynomial/polynomial.py

```diff
@@ -0,0 +1,27 @@
+import cupy
+
+
+def polyvander(x, deg):
+    """Computes the Vandermonde matrix of given degree.
+
+    Args:
+        x (cupy.ndarray): array of points
+        deg (int): degree of the resulting matrix.
+
+    Returns:
+        cupy.ndarray: The Vandermonde matrix
+
+    .. seealso:: :func:`numpy.polynomial.polynomial.polyvander`
+
+    """
+    deg.__index__()
+    if deg < 0:
+        raise ValueError('deg must be non-negative')
+    if not isinstance(x, cupy.ndarray) or x.ndim != 1:
```

Comment:
Is this condition needed?

Reply:
which one :sweat_smile: ?

Reply:
if you mean the second condition, yes it is needed because tests fail for float64 due to precision disparity if not used

Reply:
Both.
The former: This input array is expected to be a cupy.ndarray.
The latter: Please use `testing.numpy_cupy_allclose` to check array equality with tolerances.

Reply:
Why the first condition is not necessary. Yes, the input array is expected to be ndarray. What does this have to do with checking if degree is negative I didn't get that?

Reply:
``` if  not x.ndim:
       x = cupy.array(x, copy=False, ndmin=1)
```
This condition is needed for 0dim or scalar input

Reply:
`not isinstance(x, cupy.ndarray)` is always `False`, so it is not necessary.